yt_idefix/data_structures.py

```diff
@@ -519,6 +531,19 @@ def _get_constants(self, match: re.Match) -> str:
         key = match.group()
         return str(pluto_def_constants[key])
 
+    def _get_input_parameter(self, match: re.Match) -> str:
+        """Replace matched input parameter with its value"""
+        key = match.group(1)
+        if self.inifile is None:
+            self.inifile = os.path.join(self.directory, "pluto.ini")
```

Comment:
This seems fragile for the following reasons:
1) Pluto supports giving custom names for this file (`./pluto -i myfile.ini` is valid) so we can't in general assume the name of this file. Even if it is present, it might not be the one that was actually used in the simulation results we are parsing.
2) we should be able to impose when the inifile is parsed, and alleviate some of the complexity here.

because of 1) (also true for Idefix btw), I don't think we should _ever_ assume the name of the inifile, and only parse it if it was explicitly set.

Reply:
I think some default values would make it more convenient and follow the same tradition as Pluto.

Just like when users don't use a custom name for inifile, the additional option `-i myfile.ini` is not necessary and the command `./pluto` still works. 

So far this part is called _only when_ the unit definitions require the inifile. If the users use `pluto.ini` as the inifile and don't need to set it explicitly when loading the data, it still works as Pluto. Maybe it's more friendly for the Pluto users.

Reply:
I agree it's more friendly _when it works_, I'm arguing that it's more confusing _when it doesn't_. Maybe a compromise is to parse `pluto.ini` implicitly if and only if no other `*.ini` are found.

Reply:
I respect your opinion. And if we decided to parse the inifile in `self.__init__()` since then, Idefix users should also be considered. You can check my next commit :)